### PR TITLE
Sphinx

### PR DESCRIPTION
**Pull request by imagovrn**
Updated documentation:
- All new docs ported over.
- Optional build.

See https://imagovrn.github.io/flatdata/ as example. If the documentation is in acceptable state we could first push `gh-pages` branch and then submit this PR.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.9)
 project(flatdata)
```
Comment (imagovrn) [inline CMakeLists.txt]: Bumped CMake version for proper Doxygen support.

**Comment by boxdot** [comment]
I built a new docker image with your specs. It is uploaded as `boxdot/flatdata-dev:15` (the tag is this PR number). You could try to use it already in this review, if you modify the ci script. I would propose to hardcode the version of the docker image anyway and update it manually for the time being we don't have automatic builds of images.

**Comment by VeaaC** [comment]
Looks nice, I am in favour of switching

**Comment by imagovrn** [comment]
Hi @boxdot, @VeaaC. I updated the README.md to link correctly to new `.rst` files. And squashed the change to lose the noise in the git log.

**Comment by boxdot** [comment]
Looks good. -- Btw. you can squash when you merge the PR to the master. Adding smaller commits on top makes reviewing much easier since it can be done commit by commit.

**Pull request merged**